x-pack/plugins/profiling/public/components/differential_topn_functions_grid/index.tsx

```diff
@@ -0,0 +1,492 @@
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <LabelWithHint
+            label={i18n.translate('xpack.profiling.functionsView.samplesColumnLabel', {
+              defaultMessage: 'Samples',
+            })}
+            hint={i18n.translate('xpack.profiling.functionsView.samplesColumnLabel.hint', {
+              defaultMessage: 'Estimated values',
+            })}
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonSelfCPU,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <CPULabelWithHint
+            type="self"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonTotalCPU,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <CPULabelWithHint
+            type="total"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        displayAsText: 'Diff',
+        actions: { showHide: false },
+        id: TopNComparisonFunctionSortField.ComparisonDiff,
+        initialWidth: 70,
+        isSortable: false,
+      },
+    ],
+    [compareFrameAction]
+  );
+
+  const [visibleColumns, setVisibleColumns] = useState(columns.map(({ id }) => id));
+
+  function RenderCellValue({ rowIndex, columnId, setCellProps }: EuiDataGridCellValueElementProps) {
+    const isComparison = isComparisonColumn(columnId);
+    const data = isComparison ? sortedComparisonRows[rowIndex] : sortedBaseRows[rowIndex];
+
+    useEffect(() => {
+      // Add thick border to divide the baseline and comparison columns
+      if (isComparison && columnId === TopNComparisonFunctionSortField.ComparisonRank) {
+        setCellProps({
+          style: { borderLeft: theme.euiTheme.border.thick },
+        });
+      } else if (columnId === TopNFunctionSortField.TotalCPU) {
+        setCellProps({
+          style: { borderRight: theme.euiTheme.border.thin },
+        });
+      }
+    }, [columnId, isComparison, setCellProps]);
+
+    if (data === undefined) {
+      return null;
+    }
+
+    return (
+      <div data-test-subj={columnId}>
+        <FunctionRow
+          functionRow={data}
+          columnId={removeComparisonFromId(columnId)}
+          setCellProps={setCellProps}
+          totalCount={totalCount}
+          onFrameClick={onFrameClick}
+        />
+      </div>
+    );
+  }
+
+  const rowCount = Math.max(sortedBaseRows.length, sortedComparisonRows.length);
+
+  return (
+    <div>
+      <EuiDataGrid
+        data-test-subj="DiffTopNFunctions"
```

Comment:
Nit: `profilingDiffTopNFunctionsGrid` (or something similar,  I've seen this pattern across the profiling folder )